### DISCUSSION TITLE
Anki Deacks

### DISCUSSION THREAD

**Discussion by 92GC** [Deleplace/programming-idioms#232] [Ideas]
I would be nice to have automated generation of Anki decks for the idioms of each language.

**Comment by Deleplace** [comment]
I'm not familiar with Anki decks, do they require the user to type anything as an answer to be compared with the "correct" answer? Or is it learning material that gets displayed and viewed often to foster memorization?

It it makes sense for the contents of Programming-Idioms, then yeah a new endpoint could produce some deck data.

You may also try to leverage the existing JSON API (read-only), e.g. https://programming-idioms.org/api/idiom/45

**Reply by 92GC** [reply]
Yes, it is learning material that is displayed and viewed to foster memorization (with automatically adjusted spaced repetition). The idioms are already in the Question and Answer format of an Anki deck. I found programming-idioms.org from an Anki deck someone had [published](https://ankiweb.net/shared/info/2086567642). Unfortunately, that deck is out of date. Decks can either be imported with read-write access from a .apkg file or downloaded from Ankiweb with read-only access.

Copying and pasting the idioms from the website; then generating a CSV file; then converting that to a .apkg file took me about 30 minutes.
